Sources/FluentKit/Properties/Parent.swift

```diff
@@ -1,69 +1,132 @@
-        return self.$id
-    }
+    public var field: Field<To.StoredIDValue> { self.$id }
 }
 
 extension Parent: AnyProperty {
     func encode(to encoder: Encoder) throws {
         var container = encoder.singleValueContainer()
-        if let parent = self.eagerLoadedValue {
+
+        if case let .loaded(parent) = self.eagerLoadedValue {
             try container.encode(parent)
         } else {
-            try container.encode([
-                To.key(for: \._$id): self.id
-            ])
+            try container.encode([self.$id.key: self.id])
         }
     }
 
     func decode(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: _ModelCodingKey.self)
-        try self.$id.decode(from: container.superDecoder(forKey: .string(To.key(for: \._$id))))
+        try self.$id.decode(from: container.superDecoder(forKey: .string(self.$id.key)))
         // TODO: allow for nested decoding
     }
 }
 
 extension Parent: AnyField { }
+
+
+// MARK: - Specialization
+
+extension Parent where To: Model {
+    public convenience init(key: String) {
+        self.init(id: key, eagerLoadRequest: SubqueryEagerLoad(key: key))
+    }
+
+    public func query(on database: Database) -> QueryBuilder<To> {
+        return To.query(on: database).filter(\._$id == self.id)
+    }
+
+    public func get(on database: Database) -> EventLoopFuture<To> {
+        return self.query(on: database).first().flatMapThrowing { parent in
+            guard let parent = parent else {
+                throw FluentError.missingParent
+            }
+            return parent
+        }
+    }
+}
+
+extension Parent where To: OptionalType, To.Wrapped: Model, To.StoredIDValue == To.Wrapped.IDValue? {
+    public convenience init(key: String) {
+        self.init(id: key, eagerLoadRequest: SubqueryEagerLoad(key: key))
+    }
+
+    public func query(on database: Database) -> QueryBuilder<To.Wrapped> {
+        return To.Wrapped.query(on: database).filter(\To.Wrapped._$id == self.id)
+    }
+
+    public func get(on database: Database) -> EventLoopFuture<To> {
+        return self.query(on: database).first().flatMapThrowing { parent in parent as! To }
+    }
+}
+
+
+// MARK: - Optionals
+
+public protocol ParentRelatable: Codable, CustomStringConvertible {
+    associatedtype StoredIDValue: Codable, Hashable
+
+    static func eagerLoaded(for id: StoredIDValue) -> (StoredIDValue, Parent<Self>.EagerLoaded)
+
+    var storedID: StoredIDValue { get }
+}
+
+
+extension Optional: ParentRelatable, CustomStringConvertible where Wrapped: Model {
+    public typealias StoredIDValue = Wrapped.IDValue?
+
+    public var description: String { (self?.description).map { "Optional(\($0))" } ?? "nil" }
+    public var storedID: Wrapped.IDValue? { self?.id }
+
+    public static func eagerLoaded(for id: StoredIDValue) -> (StoredIDValue, Parent<Self>.EagerLoaded) {
+        return (id, id == nil ? .loaded(nil) : .notLoaded)
```

Comment:
Consider `(id, id.map { .notLoaded } ?? .loaded(nil))` to avoid having the negative condition (`nil`) as the first clause of a ternary operator expression.